src/publisher/api_v2_views.py

```diff
@@ -107,16 +110,18 @@ def article_version(request, id, version):
 
 @api_view(['POST'])
 def article_fragment(request, art_id, fragment_id):
-    only_published = is_authenticated(request)
+    if not is_authenticated(request):
+        return Response("not authenticated. only authenticated admin users can modify content", status=403)
     try:
         reserved_keys = [XML2JSON]
         ensure(fragment_id not in reserved_keys, "that key is taken")
         with transaction.atomic():
-            av = logic.most_recent_article_version(art_id, only_published)
-            data = request.data
-            frag, created, updated = fragment_logic.add(av.article, fragment_id, data, update=True)
-            ensure(created or updated, "fragment was not created/updated")
-            fragment_logic.merge_if_valid(av, quiet=False)
+            avs = logic.article_version_list(art_id, only_published=False)
+            for av in avs:
```

Comment:
So if a new version gets created it won't have that fragment, which is not what we want, I see the bug.
The original behavior was using the latest version, which is also not what we want. How do we attach the fragment to the article instead?